src/main/java/com/amazon/jenkins/ec2fleet/CloudNames.java

```diff
@@ -1,28 +1,49 @@
 package com.amazon.jenkins.ec2fleet;
 
 import jenkins.model.Jenkins;
+import org.apache.commons.lang.RandomStringUtils;
 
 import java.util.Collections;
+import java.util.HashSet;
 import java.util.Set;
 import java.util.stream.Collectors;
 
 public class CloudNames {
 
+  public static final int SUFFIX_LENGTH = 8;
+
+  private static final Set<String> usedSuffixes = new HashSet<String>();
+
   public static Boolean isUnique(final String name) {
     return !Jenkins.get().clouds.stream().anyMatch(c -> c.name.equals(name));
   }
 
-  public static String generateUnique(final String defaultName) {
+  public static String generateUnique(final String proposedName) {
     final Set<String> usedNames = Jenkins.get().clouds != null
             ? Jenkins.get().clouds.stream().map(c -> c.name).collect(Collectors.toSet())
             : Collections.emptySet();
-    String uniqName = defaultName;
 
-    int suffix = 1;
-    while (usedNames.contains(uniqName)) {
-      uniqName = defaultName + "-" + suffix++;
+    if (proposedName.equals(EC2FleetCloud.DEFAULT_FLEET_CLOUD_ID) || proposedName.equals(EC2FleetLabelCloud.DEFAULT_FLEET_CLOUD_ID) || usedNames.contains(proposedName)) {
+      return proposedName + "-" + generateSuffix();
     }
 
-    return uniqName;
+    return proposedName;
+  }
+  
+  /**
+   * We are using a randomly generated string as a suffix here because Jenkins recreates clouds on config change.
```

Comment:
Adding some more context: `Jenkins recreates clouds on config change and makes the new clouds accessible only after all clouds are replaced which makes identifying clouds with duplicate names during (re-)creation impossible, esp. for JCasC` ?